utils/handleScroll.ts

```diff
@@ -0,0 +1,74 @@
+import { ref, type Ref } from 'vue'
+
+export interface SectionInformation {
+    sectionTitle: string
+    sectionElementIdToScrollTo: string
+}
+
+export const handleScroll
+= (sectionDetailsObject: SectionInformation[], isScrolling: Ref<boolean> = ref(false), activeSection: Ref<string> = ref('')) => {
+    if (isScrolling.value) return
+    let newActiveSection: string | null = null
+
+    sectionDetailsObject.forEach(section => {
+        let rect = null
+        const foundSectionById = document.getElementById(section.sectionElementIdToScrollTo)
+        if (!foundSectionById) {
+            return
+        }
+        rect = foundSectionById.getBoundingClientRect()
+
+        const sectionId = section.sectionElementIdToScrollTo
+
+        if (rect.top <= window.innerHeight / 2 && rect.bottom >= window.innerHeight / 2) {
+            newActiveSection = sectionId
+        }
+    })
+
+    if (newActiveSection) {
+        activeSection.value = newActiveSection
+    }
+}
+
+export const scrollToSectionOfForm
+= (sectionId: string, activeSection: Ref<string> = ref('')) => {
+    document.getElementById(sectionId)?.scrollIntoView({
+        behavior: 'smooth'
+    })
+    activeSection.value = sectionId
+}
+
+export const observeFormSections = (
+    sectionsInfo: SectionInformation[],
+    isScrolling: Ref<boolean> = ref(false),
+    activeSection: Ref<string> = ref(''),
+    thresholdValue: number = 0.70
```

Comment:
this is fantastic! You nailed what I was hoping to see for how the parent uses the generic scroll 💯 

Literally the only thing I can think of at the moment is if you want to provide an override for threshold values, but we don't need that right now so no need to implement until the use case arises.

Reply:
I thought writing the parameter like this will default it to `0.70` if not provided. But maybe I am mistaken. But if so it can be updated later like you said.